### PR TITLE
Final manifest

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: docs/appendix/faqs/general.rst [hunk 1]
```diff
@@ -10,13 +10,14 @@ General
 This error is usually accompanied by more details explaining why the plugin is invalid. This is usually
 caused by an incorrectly installed plugin.
 
-If you are installing a plugin called "test", there will be a test directory under mod. In that test directory there needs to be a manifest.xml file ``/mod/test/manifest.xml``.
+If you are installing a plugin called "test", there will be a test directory under mod. In that test directory there needs to be a composer.json file ``/mod/test/composer.json``.
```
Comment (jeabakker) [inline docs/appendix/faqs/general.rst:13]: why isn't composer.json quoted

File: docs/guides/plugins.rst [hunk 2]
```diff
@@ -34,6 +35,24 @@ Besides magic constants like ``__DIR__``, its return value should not change. Th
...
+			'dependencies' => [
+				// optional list op plugin dependencies
+				'blog' => [],
+				'activity' => [
+					'position' => 'after',
+					'must_be_active' => false,
+				],
+				'file' => [
+					'position' => 'before',
+					'version' => '>2', // semver notation of required version
```
Comment (jeabakker) [inline docs/guides/plugins.rst:51]: not semver, but Composer notation

File: engine/classes/Elgg/Exceptions/Plugin/ComposerException.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Elgg\Exceptions\Plugin;
+
+use Elgg\Exceptions\PluginException;
+
+/**
+ * Indicates something wrong with the plugin composer
```
Comment (jeabakker) [inline engine/classes/Elgg/Exceptions/Plugin/ComposerException.php:8]: .json

File: engine/classes/Elgg/Exceptions/Plugin/IdMismatchException.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Elgg\Exceptions\Plugin;
+
+use Elgg\Exceptions\PluginException;
+
+/**
+ * Indicates if the plugin id mismatches
```
Comment (jeabakker) [inline engine/classes/Elgg/Exceptions/Plugin/IdMismatchException.php:8]: Indicates a mismatch between the plugin ID in the ``composer.json`` and the plugin directory

File: mod/activity/tests/phpunit/unit/Elgg/Activity/ComposerTest.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Elgg\Activity;
+
+/**
+ * @group Plugins
+ * @group Composer
+ */
+class ComposerTest extends \Elgg\Plugins\ComposerTest {
+
+}
```
Comment (jeabakker) [inline mod/activity/tests/phpunit/unit/Elgg/Activity/ComposerTest.php:11]: aaaahhhhhh!!!!!!!

File: mod/activity/tests/phpunit/unit/Elgg/Activity/StaticConfigTest.php [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Elgg\Activity;
+
+/**
+ * @group Plugins
+ * @group StaticConfig
+ */
+class StaticConfigTest extends \Elgg\Plugins\StaticConfigTest {
+
+}
```
Comment (jeabakker) [inline mod/activity/tests/phpunit/unit/Elgg/Activity/StaticConfigTest.php:11]: aaaahhhhhh!!!!!!!

File: views/default/object/plugin/elements/dependencies.php [hunk 7]
```diff
@@ -5,39 +5,60 @@
...
-
-	foreach ($columns as $column) {
-		echo "<td class=\"pas $class\">{$fields[$column]}</td>";
-	}
-
-	echo '</tr>';
+	
+	$deps_info .= '</tbody></table>';
+	
+	echo elgg_view_module('info', 'Plugin dependencies', $deps_info);
```
Comment (jeabakker) [inline views/default/object/plugin/elements/dependencies.php:57]: no elgg_echo() ????

File: views/default/resources/admin/plugin_text_file.php [hunk 8]
```diff
@@ -10,31 +10,21 @@
...
-$text_files = $plugin->getAvailableTextFiles();
-
-if (!array_key_exists($filename, $text_files)) {
-	$error = elgg_echo('admin:plugins:markdown:unknown_file');
-}
-
-$file = $text_files[$filename];
-$file_contents = file_get_contents($file);
-
+$file_contents = file_get_contents($plugin->getPath() . $filename);
```
Comment (jeabakker) [inline views/default/resources/admin/plugin_text_file.php:23]: ``$filename`` needs to be sanitized otherwise path traversal issues

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Review by jeabakker — APPROVED** [review]

**Comment by jdalsem** [comment]
going to squash merge this to resolve an unability to reword an old faulty phrased commit message... followup tickets will be created to address left over work

**Pull request merged**